### PR TITLE
Update linter

### PR DESCRIPTION
**Pull request by forbesjo**
Updated videojs-standard and added an eslintrc for editor hinting.

## Description
Updates the linter tool and add a linter dot file. Code has been updated to follow the latest linting rules.

## Requirements Checklist
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by forbesjo** [comment]
This PR is waiting on resolving some videojs-contrib-hls changes

File: src/segment-loader.js [hunk 1]
```diff
@@ -596,7 +598,7 @@ export default class SegmentLoader extends videojs.EventTarget {
     }
 
     // see if we need to begin loading immediately
-    let segmentInfo = this.checkBuffer_(this.buffered_(),
+    const segmentInfo = this.checkBuffer_(this.buffered_(),
```
Comment (mjneil) [inline src/segment-loader.js]: The arguments to this function should have another level of indentation so it lines up with `this.buffered_()`

File: src/segment-loader.js [hunk 2]
```diff
@@ -607,7 +609,7 @@ export default class SegmentLoader extends videojs.EventTarget {
       return;
     }
 
-    let isEndOfStream = detectEndOfStream(this.playlist_,
+    const isEndOfStream = detectEndOfStream(this.playlist_,
```
Comment (mjneil) [inline src/segment-loader.js]: same here with the indentation of the arguments

File: src/segment-loader.js [hunk 3]
```diff
@@ -702,7 +704,7 @@ export default class SegmentLoader extends videojs.EventTarget {
     // fetch
     if (this.fetchAtBuffer_) {
       // Find the segment containing the end of the buffer
-      let mediaSourceInfo = Playlist.getMediaInfoForTime(playlist,
+      const mediaSourceInfo = Playlist.getMediaInfoForTime(playlist,
```
Comment (mjneil) [inline src/segment-loader.js]: argument indentation

File: src/segment-loader.js [hunk 4]
```diff
@@ -711,7 +713,7 @@ export default class SegmentLoader extends videojs.EventTarget {
       startOfSegment = mediaSourceInfo.startTime;
     } else {
       // Find the segment containing currentTime
-      let mediaSourceInfo = Playlist.getMediaInfoForTime(playlist,
+      const mediaSourceInfo = Playlist.getMediaInfoForTime(playlist,
```
Comment (mjneil) [inline src/segment-loader.js]: argument indentation

File: src/sync-controller.js [hunk 5]
```diff
@@ -240,12 +240,12 @@ export default class SyncController extends videojs.EventTarget {
...
   runStrategies_(playlist, duration, currentTimeline, currentTime) {
-    let syncPoints = [];
+    const syncPoints = [];
 
     // Try to find a sync-point in by utilizing various strategies...
     for (let i = 0; i < syncPointStrategies.length; i++) {
-      let strategy = syncPointStrategies[i];
-      let syncPoint = strategy.run(this,
+      const strategy = syncPointStrategies[i];
+      const syncPoint = strategy.run(this,
```
Comment (mjneil) [inline src/sync-controller.js]: argument indentation

File: test/loader-common.js [hunk 6]
```diff
@@ -878,29 +880,29 @@ export const LoaderCommonFactory = (LoaderConstructor,
...
+      const playlist = playlistWithDuration(20);
+      const defaultGoal = Config.GOAL_BUFFER_LENGTH;
 
       Config.GOAL_BUFFER_LENGTH = 1;
       loader.playlist(playlist);
 
       loader.load();
 
-      segmentInfo = loader.checkBuffer_(videojs.createTimeRanges([[0, 1]]),
+      const segmentInfo = loader.checkBuffer_(videojs.createTimeRanges([[0, 1]]),
```
Comment (mjneil) [inline test/loader-common.js]: argument indentation

File: test/loader-common.js [hunk 7]
```diff
@@ -1033,7 +1035,7 @@ export const LoaderCommonFactory = (LoaderConstructor,
 
     QUnit.test('requests the first segment with an empty buffer', function(assert) {
 
-      let segmentInfo = loader.checkBuffer_(videojs.createTimeRanges(),
+      const segmentInfo = loader.checkBuffer_(videojs.createTimeRanges(),
```
Comment (mjneil) [inline test/loader-common.js]: argument indentation

File: test/loader-common.js [hunk 8]
```diff
@@ -1048,7 +1050,7 @@ export const LoaderCommonFactory = (LoaderConstructor,
     function(assert) {
       this.hasPlayed = false;
 
-      let segmentInfo = loader.checkBuffer_(videojs.createTimeRanges([[0, 1]]),
+      const segmentInfo = loader.checkBuffer_(videojs.createTimeRanges([[0, 1]]),
```
Comment (mjneil) [inline test/loader-common.js]: argument indentation

File: test/loader-common.js [hunk 9]
```diff
@@ -1060,13 +1062,10 @@ export const LoaderCommonFactory = (LoaderConstructor,
...
     function(assert) {
-      let buffered;
-      let segmentInfo;
-
-      buffered = videojs.createTimeRanges([
+      const buffered = videojs.createTimeRanges([
         [0, 30 + Config.GOAL_BUFFER_LENGTH]
       ]);
-      segmentInfo = loader.checkBuffer_(buffered,
+      const segmentInfo = loader.checkBuffer_(buffered,
```
Comment (mjneil) [inline test/loader-common.js]: argument indentation

File: test/loader-common.js [hunk 10]
```diff
@@ -1078,14 +1077,12 @@ export const LoaderCommonFactory = (LoaderConstructor,
...
-      let segmentInfo;
-      let playlist = playlistWithDuration(30);
+      const playlist = playlistWithDuration(30);
 
       loader.playlist(playlist);
 
-      buffered = videojs.createTimeRanges([[0, 19.999]]);
-      segmentInfo = loader.checkBuffer_(buffered,
+      const buffered = videojs.createTimeRanges([[0, 19.999]]);
+      const segmentInfo = loader.checkBuffer_(buffered,
```
Comment (mjneil) [inline test/loader-common.js]: argument indentation

File: test/loader-common.js [hunk 11]
```diff
@@ -1097,11 +1094,8 @@ export const LoaderCommonFactory = (LoaderConstructor,
     });
 
     QUnit.test('stops downloading segments at the end of the playlist', function(assert) {
-      let buffered;
-      let segmentInfo;
-
-      buffered = videojs.createTimeRanges([[0, 60]]);
-      segmentInfo = loader.checkBuffer_(buffered,
+      const buffered = videojs.createTimeRanges([[0, 60]]);
+      const segmentInfo = loader.checkBuffer_(buffered,
```
Comment (mjneil) [inline test/loader-common.js]: argument indentation

File: test/loader-common.js [hunk 12]
```diff
@@ -1113,14 +1107,11 @@ export const LoaderCommonFactory = (LoaderConstructor,
...
-      let segmentInfo;
-      let playlist;
+      const buffered = videojs.createTimeRanges([[0, 59.9]]);
+      const playlist = playlistWithDuration(60);
 
-      buffered = videojs.createTimeRanges([[0, 59.9]]);
-      playlist = playlistWithDuration(60);
       playlist.segments[playlist.segments.length - 1].end = 59.9;
-      segmentInfo = loader.checkBuffer_(buffered,
+      const segmentInfo = loader.checkBuffer_(buffered,
```
Comment (mjneil) [inline test/loader-common.js]: argument indentation

**Comment by forbesjo** [comment]
This PR will wait until other major work has finised

**Comment by forbesjo** [comment]
Going to close this an open a chore to update linting

**Pull request closed without merging**